tests/src/whisk/core/database/test/ReplicatorTests.scala

```diff
@@ -0,0 +1,280 @@
+        println(s"Running replicator: $sourceDbUrl, $targetDbUrl, $dbPrefix, $expires, $continuous")
+
+        val continuousFlag = if (continuous) Some("--continuous") else None
+        val cmd = Seq(python, replicator, "--sourceDbUrl", sourceDbUrl, "--targetDbUrl", targetDbUrl, "replicate", "--dbPrefix", dbPrefix, "--expires", expires.toSeconds.toString) ++ continuousFlag
+        val rr = TestUtils.runCmd(0, new File("."), cmd: _*)
+
+        val Seq(created, deleted) = Seq("create backup: ", "deleting backup: ").map { prefix =>
+            rr.stdout.lines.collect {
+                case line if line.startsWith(prefix) => line.replace(prefix, "")
+            }.toList
+        }
+
+        println(s"Created: $created")
+        println(s"Deleted: $deleted")
+
+        (created, deleted)
+    }
+
+    /** Runs the replicator script to replay databases */
+    def runReplay(sourceDbUrl: String, targetDbUrl: String, dbPrefix: String) = {
+        println(s"Running replay: $sourceDbUrl, $targetDbUrl, $dbPrefix")
+        val rr = TestUtils.runCmd(0, new File("."), WhiskProperties.python, WhiskProperties.getFileRelativeToWhiskHome("tools/db/replicateDbs.py").getAbsolutePath, "--sourceDbUrl", sourceDbUrl, "--targetDbUrl", targetDbUrl, "replay", "--dbPrefix", dbPrefix)
+
+        val line = """(\w+) -> (\w+) \((\w+)\)""".r.unanchored
+        val replays = rr.stdout.lines.collect {
+            case line(backup, target, id) => (backup, target, id)
+        }.toList
+
+        println(s"Replays created: $replays")
+
+        replays
+    }
+
+    /** Wait for a replication to finish */
+    def waitForReplication(dbName: String) = {
+        waitfor(() => {
+            val replicatorDoc = replicatorClient.getDoc(dbName).futureValue
+            replicatorDoc shouldBe 'right
+
+            val state = replicatorDoc.right.get.fields.get("_replication_state")
+            println(s"Waiting for replication, state: $state")
+
+            state.contains("completed".toJson)
+        }, totalWait = 2.minutes)
+    }
+
+    /** Wait for database to appear */
+    def waitForDatabase(dbName: String) = {
+        val client = new ExtendedCouchDbRestClient(config.dbProtocol, config.dbHost, config.dbPort.toInt, config.dbUsername, config.dbPassword, dbName)
+        waitfor(() => {
+            client.getAllDocs(includeDocs = Some(true)).futureValue.isRight
+        })
+        client
+    }
+
+    /** Wait for a document to appear */
+    def waitForDocument(client: ExtendedCouchDbRestClient, id: String) = waitfor(() => client.getDoc(id).futureValue.isRight)
+
+    /** Compares to databases to full equality */
+    def compareDatabases(dbNames: Seq[String]) = {
+        val dbDocuments = dbNames.map { dbName =>
+            val client = new ExtendedCouchDbRestClient(config.dbProtocol, config.dbHost, config.dbPort.toInt, config.dbUsername, config.dbPassword, dbName)
+            val documents = client.getAllDocs(includeDocs = Some(true)).futureValue
```

Comment:
should put a comment somewhere here that this should not be used to test large databases.